### PR TITLE
NEXUS-21942 Implemented Routing Rules support in Conan plug-in

### PR DESCRIPTION
**Pull request by igorudovika**
NEXUS-21942 Implemented Routing Rules support in Conan plug-in

### CODE REVIEW

File: nexus-repository-conan-it/src/test/java/org/sonatype/nexus/plugins/conan/internal/ConanRoutingRuleIT.java [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+  @Inject
+  private RoutingRuleStore ruleStore;
+
+  @Rule
+  public RepositoryRuleConan repos = new RepositoryRuleConan(() -> repositoryManager);
+
+  @Rule
+  public RoutingRuleRule routingRules = new RoutingRuleRule(() -> ruleStore);
+
+  private EntityId createBlockedRoutingRule(final String name, final String matcher) {
```
Comment (DreamVan) [inline nexus-repository-conan-it/src/test/java/org/sonatype/nexus/plugins/conan/internal/ConanRoutingRuleIT.java]: trivial: please move methods after test's configuration methods

File: nexus-repository-conan-it/src/test/java/org/sonatype/nexus/plugins/conan/internal/ConanRoutingRuleIT.java [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+  {
+    try (CloseableHttpResponse response = client.get(path)) {
+      StatusLine statusLine = response.getStatusLine();
+      assertThat("Repository:" + repository.getName() + " Path:" + path, statusLine.getStatusCode(), is(responseCode));
+    }
+  }
+
+  @Test
+  public void testRoutingRule() throws Exception {
+    assertGetResponseStatus(proxyClient, proxyRepo, JSON_FOR_MODERN_CPP_URL, 200);
```
Comment (proteus1121) [inline nexus-repository-conan-it/src/test/java/org/sonatype/nexus/plugins/conan/internal/ConanRoutingRuleIT.java]: Maybe use javax.ws.rs.core.Response.Status.OK.getStatusCode()? and for 403 below?

File: nexus-repository-conan-it/src/test/java/org/sonatype/nexus/plugins/conan/internal/ConanRoutingRuleIT.java (hunk 1 above)
Comment (igorudovika) [reply nexus-repository-conan-it/src/test/java/org/sonatype/nexus/plugins/conan/internal/ConanRoutingRuleIT.java]: done df0382fe1e4b0956e9b0122fcf65fbfa08a42b56

File: nexus-repository-conan-it/src/test/java/org/sonatype/nexus/plugins/conan/internal/ConanRoutingRuleIT.java (hunk 2 above)
Comment (igorudovika) [reply nexus-repository-conan-it/src/test/java/org/sonatype/nexus/plugins/conan/internal/ConanRoutingRuleIT.java]: 574f250fe13f0b7c107dce9df11b50e597f7bcaf 
Is it better now?

Comment (proteus1121) [reply nexus-repository-conan-it/src/test/java/org/sonatype/nexus/plugins/conan/internal/ConanRoutingRuleIT.java]: I think yes

**Pull request merged**